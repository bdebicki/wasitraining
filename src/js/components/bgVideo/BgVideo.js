import { BG } from '../../elementHandlers/elementHandlers';
import LAYOUT from '../../elementHandlers/layout';
import { addVideo } from '../../utils/addElement';

export default class BgVideo {
	constructor(target) {
		this.target = target;
	}

	static renderVideoContainer() {
		const div = document.createElement('div');

		div.id = LAYOUT.BG_VIDEO_ID;

		return div;
	}

	static renderVideo() {
		const settings = {
			classNames: BG.VIDEO_CLASS,
			src: '/videos/bg-video.webm',
			placeholder: '/images/bg-img.png',
			width: '1366',
			height: '768',
		};

		return addVideo(settings);
	}

	static renderImg() {
		const img = document.createElement('img');

		img.classList.add(BG.VIDEO_CLASS);
		img.src = '/images/bg-img.png';
		img.width = '1366';
		img.height = '768';

		return img;
	}

	render() {
		const videoBlock = BgVideo.renderVideoContainer();

		videoBlock.appendChild(BgVideo.renderVideo());

		document.getElementById(this.target).appendChild(videoBlock);
	}
}
