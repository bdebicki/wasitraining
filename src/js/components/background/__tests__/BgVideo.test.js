import BgVideo from '../BgVideo';

describe('test background video component', () => {
	it('render background video component', () => {
		// when
		const videoEl = BgVideo.render();

		// then
		expect(videoEl).toMatchSnapshot();
	});
});
