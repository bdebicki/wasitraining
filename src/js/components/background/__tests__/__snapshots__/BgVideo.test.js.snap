// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test background video component render background video component 1`] = `
<div
  id="bgVideo"
>
  <video
    autoplay=""
    class="bgVideo__video"
    height="768"
    loop=""
    poster="/images/bg-img.png"
    src="/videos/bg-video.webm"
    width="1366"
  />
</div>
`;
